89.md

```diff
@@ -0,0 +1,112 @@
+NIP-89
+======
+
+Recommended Application Handlers
+--------------------------------
+
+`draft` `optional` `author:pablof7z`
+
+This NIP describes `kind:31989` and `kind:31990`: a way to discover applications that can handle unknown event-kinds.
+
+## Rationale
+Nostr's discoverability and transparent event interaction is one of its most interesting/novel mechanics.
+This NIP provides a simple way for clients to discover applications that handle events of a specific kind to ensure smooth cross-client and cross-kind interactions.
+
+### Parties involved
+There are three actors to this workflow:
+
+* application that handles a specific event kind (note that an application doesn't necessarily need to be a distinct entity and it could just be the same pubkey as user A)
+    * Publishes `kind:31990`, detailing how apps should redirect to it
+* user A, who recommends an app that handles a specific event kind
+    * Publishes `kind:31989`
+* user B, who seeks a recommendation for an app that handles a specific event kind
+    * Queries for `kind:31989` and, based on results, queries for `kind:31990`
+
+# Events
+
+## Recommendation event
+```json
+{
+    "kind": 31989,
+    "pubkey": <recommender-user-pubkey>,
+    "tags": [
+        [ "d", <supported-event-kind> ],
```

Comment:
We could comment that we're using d tag as kind, instead of separate k tag, so make sure user only has on app list per kind. Or, if it's not our intention, then we should consider not assigning special meaning to d-tag and using k tag instead.

Reply:
no, the d tag should be the kind to prevent multiple events recommending the same kind which would be confusing